langchain/src/document_loaders/github.ts

```diff
@@ -75,48 +101,52 @@ export class GithubRepoLoader extends BaseDocumentLoader {
               new Document({ pageContent: fileContent, metadata })
             );
           } catch (e) {
-            this.handleError(`Failed to fetch file content: ${file.path}`);
+            this.handleError(
+              `Failed to fetch file content: ${file.path}, ${e}`
+            );
           }
         }
       }
     } catch (error) {
-      this.handleError(`Failed to process directory: ${path}`);
+      this.handleError(`Failed to process directory: ${path}, ${error}`);
     }
   }
 
-  private async fetchRepoFiles(path: string): Promise<any[]> {
+  private async fetchRepoFiles(path: string): Promise<GithubFile[]> {
     const url = `https://api.github.com/repos/${this.owner}/${this.repo}/contents/${path}?ref=${this.branch}`;
     const response = await fetch(url);
```

Comment:
If we wanted to be really nice here we could do an exponential backoff retry, we have this to make it painless https://github.com/hwchase17/langchainjs/blob/main/langchain/src/util/async_caller.ts#L22

Reply:
I'll create a separate issue for us to use this in all document loaders / tool, everywhere use fetch basically